extensions/sql-migration/src/wizard/loginSelectorPage.ts

```diff
@@ -327,43 +444,104 @@ export class LoginSelectorPage extends MigrationWizardPage {
-			await this.updateValuesOnSelection();
-		}));
+		return table;
+	}
 
-		// load unfiltered table list and pre-select list of logins saved in state
-		await this._filterTableList('', this.migrationStateModel._loginMigrationModel.loginsForMigration);
+	private _createSystemLoginTablesTable(view: azdata.ModelView): azdata.TableComponent {
+		const cssClass = 'no-borders';
+		const table = this._view.modelBuilder.table()
+			.withProps({
+				data: [],
+				width: 650,
+				height: '600px',
+				display: 'flex',
+				forceFitColumns: azdata.ColumnSizingMode.ForceFit,
+				columns: [
+					{
+						name: constants.SOURCE_LOGIN,
+						value: 'sourceLogin',
+						type: azdata.ColumnType.text,
+						width: 250,
+						cssClass: cssClass,
+						headerCssClass: cssClass,
+					},
+					{
+						name: constants.LOGIN_TYPE,
+						value: 'loginType',
+						type: azdata.ColumnType.text,
+						width: 90,
+						cssClass: cssClass,
+						headerCssClass: cssClass,
+					},
+					{
+						name: constants.DEFAULT_DATABASE,
+						value: 'defaultDatabase',
+						type: azdata.ColumnType.text,
+						width: 130,
+						cssClass: cssClass,
+						headerCssClass: cssClass,
+					},
+					{
+						name: constants.LOGIN_STATUS_COLUMN,
+						value: 'status',
+						type: azdata.ColumnType.text,
+						width: 90,
+						cssClass: cssClass,
+						headerCssClass: cssClass,
+					},
+					<azdata.HyperlinkColumn>{
+						name: constants.LOGIN_TARGET_STATUS_COLUMN,
+						value: 'targetStatus',
+						width: 150,
+						type: azdata.ColumnType.hyperlink,
+						icon: IconPathHelper.inProgressMigration,
+						showText: true,
+						cssClass: cssClass,
+						headerCssClass: cssClass,
+					},
+				],
+				CSSStyles: { 'margin-top': '8px' }
+			})
+			.component();
 
-		const flex = view.modelBuilder.flexContainer().withLayout({
-			flexFlow: 'column',
-			height: '100%',
-		}).withProps({
-			CSSStyles: {
-				'margin': '-20px 28px 0px 28px'
-			}
-		}).component();
-		flex.addItem(this._windowsAuthInfoBox, { flex: '0 0 auto' });
-		flex.addItem(refreshContainer, { flex: '0 0 auto' });
-		flex.addItem(this.createSearchComponent(), { flex: '0 0 auto' });
-		flex.addItem(this._loginCount, { flex: '0 0 auto' });
-		flex.addItem(this._loginSelectorTable);
-		flex.addItem(this.createAadDomainNameComponent(), { flex: '0 0 auto', CSSStyles: { 'margin-top': '8px' } });
-		return flex;
+		return table;
 	}
 
 	private async _getSourceLogins() {
 		const stateMachine: MigrationStateModel = this.migrationStateModel;
-		const sourceLogins: LoginTableInfo[] = [];
+		var sourceLogins: LoginTableInfo[] = [];
 
 		// execute a query against the source to get the logins
 		try {
 			sourceLogins.push(...await collectSourceLogins(
 				await getSourceConnectionId(),
 				stateMachine.isWindowsAuthMigrationSupported));
+
+			// validate Login Eligibility result contains system logins in Exception map from which system login names can be extracted.
```

Comment:
validation and separation code is repeated in both files , can we have a function for this?

Reply:
done